sechub-wrapper-owasp-zap/src/main/java/com/mercedesbenz/sechub/owaspzapwrapper/scan/AbstractScan.java

```diff
@@ -0,0 +1,364 @@
+        registerUrlsIncludedInContext();
+        registerUrlsExcludedFromContext();
+    }
+
+    /**
+     * Wait for the results of the spider. Periodically checks the progress of the
+     * spider.
+     *
+     * @param response
+     * @throws ClientApiException
+     */
+    protected void waitForSpiderResults(ApiResponse response) throws ClientApiException {
+        String scanID = ((ApiResponseElement) response).getValue();
+        int progressSpider = 0;
+
+        long startTime = System.currentTimeMillis();
+        long maxDuration = scanTimingHelper.computeSpiderMaxScanDuration(scanConfig.isActiveScanEnabled(), scanConfig.isAjaxSpiderEnabled(), remainingScanTime);
+
+        boolean timeOut = false;
+
+        while (progressSpider < 100 && !timeOut) {
+            waitForNextCheck();
+            progressSpider = Integer.parseInt(((ApiResponseElement) clientApi.spider.status(scanID)).getValue());
+            LOG.info("For scan {}: Spider progress {}%", scanConfig.getContextName(), progressSpider);
+
+            if (scanConfig.isVerboseOutput()) {
+                List<ApiResponse> spiderResults = ((ApiResponseList) clientApi.spider.results(scanID)).getItems();
+                for (ApiResponse result : spiderResults) {
+                    LOG.info("For scan {}: Result: {}", scanConfig.getContextName(), result.toString());
+                }
+            }
+            timeOut = System.currentTimeMillis() - startTime > maxDuration;
+        }
+        /* stop spider - otherwise running in background */
+        clientApi.spider.stop(scanID);
+
+        LOG.info("For scan {}: Spider completed.", scanConfig.getContextName());
+        remainingScanTime = remainingScanTime - (System.currentTimeMillis() - startTime);
+    }
+
+    /**
+     * Wait for the results of the ajax spider. Periodically checks the progress of
+     * the ajax spider.
+     *
+     * @param response
+     * @throws ClientApiException
+     */
+    protected void waitForAjaxSpiderResults(ApiResponse response) throws ClientApiException {
+        String ajaxSpiderStatus = null;
+
+        long startTime = System.currentTimeMillis();
+        long maxDuration = scanTimingHelper.computeAjaxSpiderMaxScanDuration(scanConfig.isActiveScanEnabled(), remainingScanTime);
+
+        boolean timeOut = false;
+
+        while (!isAjaxSpiderStopped(ajaxSpiderStatus) && !timeOut) {
+            waitForNextCheck();
+            ajaxSpiderStatus = ((ApiResponseElement) clientApi.ajaxSpider.status()).getValue();
+            LOG.info("For scan {}: AjaxSpider status {}", scanConfig.getContextName(), ajaxSpiderStatus);
+
+            if (scanConfig.isVerboseOutput()) {
+                String numberOfResults = ((ApiResponseElement) clientApi.ajaxSpider.numberOfResults()).getValue();
+                List<ApiResponse> spiderResults = ((ApiResponseList) clientApi.ajaxSpider.results("0", numberOfResults)).getItems();
```

Comment:
What is this magic "0" ? If it is a magic constant, we should provide a constant in java as well + dedicated name so it's clear what is meant.

Reply:
I added a comment to this section of the code. The "0" is the index, because if verbose output is enabled we log the currently available results. The ZAP API uses a String here, but it is index 0 to mark start with the first and log all currently available.